scheduling/first_come_first_served.py

```diff
@@ -1,66 +1,83 @@
 # Implementation of First Come First Served CPU scheduling Algorithm
 # In this Algorithm we just care about the order that the processes arrived
 # without carring about their duration time
-
-# Function to calculate the waiting time of each process
-
-
-def findWaitingTime(processes, n, duration_time, waiting_time):
-
-    # Initialising the first prosecces' waiting time with 0
-    waiting_time[0] = 0
-
-    # calculating waiting time
-    for i in range(1, n):
+# https://en.wikipedia.org/wiki/Scheduling_(computing)#First_come,_first_served
+from typing import List
+from functools import reduce
+
+
+def calculate_waiting_time(
+    number_of_processes: int, duration_time: List[int]
+) -> List[int]:
+    """
+    This function calculates the waiting time of some processes that have a specified duration time.
+        Return: The waiting time for each process.
+    >>> calculate_waiting_time(3, [5,10,15])
+    [0, 5, 15]
+    >>> calculate_waiting_time(5, [1,2,3,4,5])
+    [0, 1, 3, 6, 10]
+    >>> calculate_waiting_time(2, [10, 3])
+    [0, 10]
+    """
+    # Initialize the result of the waiting time to all zeroes
+    waiting_time: List[int] = [0] * number_of_processes
+    # calculating waiting time. First item should always be 0 so the iteration starts from index 1
+    for i in range(1, number_of_processes):
         waiting_time[i] = duration_time[i - 1] + waiting_time[i - 1]
+    return waiting_time
+
+
+def calculate_turnaround_time(
+    number_of_processes: int, duration_time: List[int], waiting_time: List[int]
+) -> List[int]:
+    """
+    This function calculates the turnaround time of some processes.
+        Return: The time difference between the completion time and the arrival time.
+                Practically waiting_time + duration_time
+    >>> calculate_turnaround_time(3, [5, 10, 15], [0, 5, 15])
+    [5, 15, 30]
+    >>> calculate_turnaround_time(5, [1, 2, 3, 4, 5], [0, 1, 3, 6, 10])
+    [1, 3, 6, 10, 15]
+    >>> calculate_turnaround_time(2, [10, 3], [0, 10])
+    [10, 13]
+    """
+    # Initialize the result of the waiting time to all zeroes
+    turnaround_time: List[int] = [0] * number_of_processes
+    for i in range(number_of_processes):
+        turnaround_time[i] = duration_time[i] + waiting_time[i]
+    return turnaround_time
```

Comment:
List comprehension: `return [duration_time[i] + waiting_time[i] for i in len(duration_time)]